addons/mail/controllers/discuss/gif.py

```diff
@@ -0,0 +1,58 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import json
+import requests
+from odoo.http import request, route, Controller
+
+
+class DiscussGifController(Controller):
+
+    def _request_gifs(self, url):
+        try:
+            response = requests.get(url, timeout=3)
+        except requests.exceptions.RequestException:
+            return False
+        if response.status_code != requests.codes.ok:
+            return False
+        return response
+
+    @route("/discuss/gif/search", type="json", auth="public", cors="*")
+    def search(self, search_term, locale="en", country="US", position=False):
+        config = request.env["ir.config_parameter"].sudo().tenor_query_params()
+        url = f"https://tenor.googleapis.com/v2/search?q={search_term}&key={config.get('api_key')}&client_key={config.get('client_key')}&limit={config.get('tenor_gif_limit')}&contentfilter={config.get('content_filter')}&locale={locale}&country={country}&media_filter=tinygif"
+        if position:
+            url = f"{url}&pos={position}"
+        response = self._request_gifs(url)
+        if response:
+            return json.loads(response.content)
+
+    @route("/discuss/gif/categories", type="json", auth="public", cors="*")
+    def categories(self, locale="en", country="US"):
+        config = request.env["ir.config_parameter"].sudo().tenor_query_params()
+        url = f"https://tenor.googleapis.com/v2/categories?key={config.get('api_key')}&client_key={config.get('client_key')}&contentfilter={config.get('content_filter')}&locale={locale}&country={country}"
+        response = self._request_gifs(url)
+        if response:
+            return json.loads(response.content)
+
+    @route("/discuss/gif/add_favorite", type="json", auth="user")
+    def add_favorite(self, tenor_gif_id):
+        request.env["discuss.gif.favorite"].create({"tenor_gif_id": tenor_gif_id})
+
+    def _gif_posts(self, ids):
```

Comment:
This method convert the favorite to gifs from the tenor API, we would still need it.
It's called posts because it's the name of the endpoint, but that could need improvement.